### PR TITLE
refactor(Entityframework, Caller, Cache): Optimize performance

### PR DESCRIPTION
**Pull request by zhenlei520**
1. perf(EntityFrameworkCore): Optimize MasaDbContext extension experience
2. perf(caller): Strengthen Caller Name assignmentand and Optimize error prompts #20 
3. feat(cache): Support to get configuration from IConfiguration to add Redis
4. chore(dapr): Modify log information and exception prompt

### CODE REVIEW

File: src/Development/Masa.Utils.Development.Dapr.AspNetCore/DaprBackgroundOptions.cs [hunk 1]
```diff
@@ -30,7 +30,7 @@ public int Time
         set
         {
             if (value <= 0)
-                throw new ArgumentException($"{nameof(MaxRetryTimes)} Must be greater than 0");
+                throw new ArgumentException($"{nameof(Time)} Must be greater than 0");
```
Comment (doddgu) [inline src/Development/Masa.Utils.Development.Dapr.AspNetCore/DaprBackgroundOptions.cs:33]: refactor

File: test/Masa.Utils.Caller.Core.Tests/CallerTest.cs [hunk 2]
```diff
@@ -44,4 +42,135 @@ public void TestCaller()
...
+    public void TestRepeatCallerName()
+    {
+        IServiceCollection services = new ServiceCollection();
+        Assert.ThrowsException<ArgumentException>(() =>
+        {
+            services.AddCaller(opt =>
+            {
+                opt.UseHttpClient(builder =>
+                {
+                    builder.Name = "gitee";
```
Comment (doddgu) [inline test/Masa.Utils.Caller.Core.Tests/CallerTest.cs:107]: gitee?

File: test/Masa.Utils.Caller.Core.Tests/CallerTest.cs [hunk 3]
```diff
@@ -44,4 +42,135 @@ public void TestCaller()
...
+    public void TestRepeatCallerName2()
+    {
+        IServiceCollection services = new ServiceCollection();
+        Assert.ThrowsException<ArgumentException>(() =>
+        {
+            services.AddCaller(opt =>
+            {
+                opt.UseHttpClient(builder =>
+                {
+                    builder.Name = "gitee";
```
Comment (doddgu) [inline test/Masa.Utils.Caller.Core.Tests/CallerTest.cs:131]: gitee?

File: src/Development/Masa.Utils.Development.Dapr.AspNetCore/DaprBackgroundOptions.cs (hunk 1 above)
Comment (zhenlei520) [reply src/Development/Masa.Utils.Development.Dapr.AspNetCore/DaprBackgroundOptions.cs]: processed

File: test/Masa.Utils.Caller.Core.Tests/CallerTest.cs (hunk 2 above)
Comment (zhenlei520) [reply test/Masa.Utils.Caller.Core.Tests/CallerTest.cs]: Here is the name of the caller, which has no practical effect. For the convenience of understanding, it was renamed giteeCaller

File: test/Masa.Utils.Caller.Core.Tests/CallerTest.cs (hunk 3 above)
Comment (zhenlei520) [reply test/Masa.Utils.Caller.Core.Tests/CallerTest.cs]: Here is the name of the caller, which has no practical effect. For the convenience of understanding, it was renamed giteeCaller

File: test/Masa.Utils.Caller.Core.Tests/CallerTest.cs [hunk 4]
```diff
@@ -44,4 +42,135 @@ public void TestCaller()
...
+    public void TestRepeatCallerName()
+    {
+        IServiceCollection services = new ServiceCollection();
+        Assert.ThrowsException<ArgumentException>(() =>
+        {
+            services.AddCaller(opt =>
+            {
+                opt.UseHttpClient(builder =>
+                {
+                    builder.Name = "giteeCaller";
```
Comment (doddgu) [inline test/Masa.Utils.Caller.Core.Tests/CallerTest.cs:107]: gitee?

File: test/Masa.Utils.Caller.Core.Tests/CallerTest.cs [hunk 5]
```diff
@@ -44,4 +42,135 @@ public void TestCaller()
...
+                    builder.BaseApi = "https://github.com/masastack";
+                    builder.IsDefault = true;
+                });
+            });
+
+            services.AddCaller(opt =>
+            {
+                opt.UseHttpClient(builder =>
+                {
+                    builder.Name = "gitee";
```
Comment (doddgu) [inline test/Masa.Utils.Caller.Core.Tests/CallerTest.cs:169]: gitee?

File: test/Masa.Utils.Caller.Core.Tests/CallerTest.cs [hunk 6]
```diff
@@ -44,4 +42,135 @@ public void TestCaller()
...
+    public void TestRepeatCallerName2()
+    {
+        IServiceCollection services = new ServiceCollection();
+        Assert.ThrowsException<ArgumentException>(() =>
+        {
+            services.AddCaller(opt =>
+            {
+                opt.UseHttpClient(builder =>
+                {
+                    builder.Name = "giteeCaller";
```
Comment (doddgu) [inline test/Masa.Utils.Caller.Core.Tests/CallerTest.cs:131]: gitee?

File: test/Masa.Utils.Caller.Core.Tests/CallerTest.cs [hunk 7]
```diff
@@ -44,4 +42,135 @@ public void TestCaller()
...
+                    builder.BaseApi = "https://github.com/masastack";
+                    builder.IsDefault = true;
+                });
+                opt.UseHttpClient(builder =>
+                {
+                    builder.Name = "giteeCaller";
```
Comment (doddgu) [inline test/Masa.Utils.Caller.Core.Tests/CallerTest.cs:113]: gitee?

File: test/Masa.Utils.Caller.Core.Tests/CallerTest.cs (hunk 4 above)
Comment (zhenlei520) [reply test/Masa.Utils.Caller.Core.Tests/CallerTest.cs]: processed

File: test/Masa.Utils.Caller.Core.Tests/CallerTest.cs (hunk 5 above)
Comment (zhenlei520) [reply test/Masa.Utils.Caller.Core.Tests/CallerTest.cs]: processed

File: test/Masa.Utils.Caller.Core.Tests/CallerTest.cs (hunk 6 above)
Comment (zhenlei520) [reply test/Masa.Utils.Caller.Core.Tests/CallerTest.cs]: processed

File: test/Masa.Utils.Caller.Core.Tests/CallerTest.cs (hunk 7 above)
Comment (zhenlei520) [reply test/Masa.Utils.Caller.Core.Tests/CallerTest.cs]: processed

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**